### PR TITLE
Add market closure governance proposal

### PR DESCRIPTION
**Pull request by witgaw**

### CODE REVIEW

**Review by witgaw — COMMENTED** [review]
Ready for review

File: protocol/0028-GOVE-governance.md [hunk 1]
```diff
@@ -263,6 +263,17 @@ Enactment of an asset modification proposal is:
 
 ## 5. Transfers initiated by Governance (post Oregon trail)
 
+## 6. Close market
+
+Any type of market (either fixed expiry market or perpetual) can be closed via a governance vote.
+
+A proposal to close a market contains:
+
+1. final settlement price
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md]: I wonder how to handle this for a spot market (which we want to allow to be closed) but it won't need a final settlement price.

File: protocol/0028-GOVE-governance.md [hunk 2]
```diff
@@ -263,6 +263,17 @@ Enactment of an asset modification proposal is:
 
 ## 5. Transfers initiated by Governance (post Oregon trail)
 
+## 6. Close market
+
+Any type of market (either fixed expiry market or perpetual) can be closed via a governance vote.
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md]: We should make clear that it's the same voting rules as market update proposals (i.e. LPs vote via ELS, can be overriden by governance token holders).

**Review by davidsiska-vega — COMMENTED** [review]
I've left two comments.

File: protocol/0028-GOVE-governance.md (hunk 1 above)
Comment (witgaw) [reply protocol/0028-GOVE-governance.md]: We can either build more complex validation where market validates proposal that's meant to apply to it or just always require some price, but spot would ignore it (not elegant, but easiest to implement)

File: protocol/0028-GOVE-governance.md (hunk 2 above)
Comment (witgaw) [reply protocol/0028-GOVE-governance.md]: I remember that being mentioned last time we discussed it, but couldn't find anything in the specs about it (probably was looking in a wrong place)

File: protocol/0028-GOVE-governance.md [hunk 3]
```diff
@@ -355,7 +355,37 @@ All in memory active governance transfers must be included in the snapshot of th
...
+### 6.1. Move market to a closed state
+
+Any type of market (either fixed expiry market or perpetual) can be closed via a governance vote.
+
+A proposal to close a market contains:
+
+1. an enactment time
+1. final settlement price (not required for spot markets)
+
+Once market is closed the process cannot be reversed.
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md:380]: Do the market change proposal voting rules apply? I would suggest they do. 

Btw. on that we have that only an LP with a certian ELS can make the proposal (min tokens *and* min ELS met). I think we should change those to (min tokens *OR* min ELS met).

File: protocol/0028-GOVE-governance.md [hunk 4]
```diff
@@ -470,7 +500,15 @@ APIs should also exist for clients to:
...
 - In particular a market change proposal that's to modify the parent market on a market in `pending` state (i.e. voting has successfully completed and the market is in the opening auction) will be accepted and if it's the enactment time happens to be before the opening auction ends then the parent is used (assuming the proposed parent doesn't already have a successor). (<a name="0028-GOVE-071" href="#0028-GOVE-071">0028-GOVE-071</a>)
 - A market change that's to modify any parameters on a market in `pending` state (i.e. voting has successfully completed on the market creation and the market is in the opening auction) will run voting rules the same as market creation proposals i.e. LPs don't get a vote. (<a name="0028-GOVE-072" href="#0028-GOVE-072">0028-GOVE-072</a>)
-
+- A governance proposal to close a market which doesn't specify the final settlement price gets rejected buy the markets which require it (non-spot). (<a name="0028-GOVE-108" href="#0028-GOVE-108">0028-GOVE-108</a>)
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md]:
```suggestion
- A governance proposal to close a market which doesn't specify the final settlement price gets rejected by the markets which require it (non-spot). (<a name="0028-GOVE-108" href="#0028-GOVE-108">0028-GOVE-108</a>)
```

File: protocol/0028-GOVE-governance.md [hunk 5]
```diff
@@ -470,7 +500,15 @@ APIs should also exist for clients to:
...
+- When multiple market closure governance proposals are submitted and accepted then one to arrive last is the one whose enactment time and final settlement price get used to close the market (<a name="0028-GOVE-109" href="#0028-GOVE-109">0028-GOVE-109</a>)
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md]:
```suggestion
- When multiple market closure governance proposals are submitted and accepted then one to be approved by governance last (i.e latest closing date and vote goes through) is the one whose enactment time and final settlement price get used to close the market (<a name="0028-GOVE-109" href="#0028-GOVE-109">0028-GOVE-109</a>)
```

File: protocol/0043-MKTL-market_lifecycle.md [hunk 6]
```diff
@@ -295,3 +295,26 @@ After `market.liquidity.successorLaunchWindowLength` has passed since market set
...
 1. The market state is set to cancelled.
+
+### Market gets closed via a governance proposal (<a name="0043-MKTL-004" href="#0043-MKTL-004">0043-MKTL-004</a>)
+
+1. Once the governance proposal to close the market gets enacted any auction that the market may be in gets uncrossed and trades get generated.
+1. All the other orders are cancelled and no further trades get generated.
+1. Any new orders get rejected.
+1. Liquidity commitments cannot be modified or cancelled.
+1. Final settlement is carried out and the transfers reflect the difference in the last mark price and the final settlement price supplied by the governance proposal.
+1. All the funds from market specific accounts get released to appropriate accounts.
```
Comment (davidsiska-vega) [inline protocol/0043-MKTL-market_lifecycle.md]:
```suggestion
1. All the funds from market specific accounts get released to appropriate accounts; the insurance pool perhaps after the delay to allow for transfer into a successor market.
```

File: protocol/0028-GOVE-governance.md (hunk 3 above)
Comment (witgaw) [reply protocol/0028-GOVE-governance.md:380]: Clarified now, happy to add the modification you mentioned, but perhaps in a separate PR?

File: protocol/0028-GOVE-governance.md [hunk 7]
```diff
@@ -141,10 +143,12 @@ Note: see below for details on minimum participation rate and minimum required m
 
 Not in scope: minimum participation of active users, i.e. 90% of the _active_ users of the vega network have to take part in the vote. Minimum participation is currently always measured against the total possible participation.
 
+### Market change proposal outcome
+
 For market change proposals the network will additionally calculate
 
 1. `LP participation rate = SUM (equity-like share of all LPs who cast a vote)` (no need to divide by anything as equity-like share sums up to `1`).
-1. `LP for rate = SUM (all who voted for) / LP participation rate`.
+1. `LP for rate = SUM (equity-like share of all LPs who cast a for vote)) / LP participation rate`.
```
Comment (davidsiska-vega) [inline protocol/0028-GOVE-governance.md]:
```suggestion
1. `LP for rate = SUM (equity-like share of all LPs who cast a for vote))`.
```

File: protocol/0028-GOVE-governance.md (hunk 3 above)
Comment (davidsiska-vega) [reply protocol/0028-GOVE-governance.md:380]: Let's do separate PR as it will be cleaner to track implementation.

**Review by davidsiska-vega — DISMISSED** [review]

**Review by davidsiska-vega — APPROVED** [review]

**Pull request merged**